### PR TITLE
[FEAT] update to ethers-v6

### PR DESCRIPTION
**Pull request by RauppRafael**

### CODE REVIEW

File: src/classes/WalletProvider/WalletProvider.ts [hunk 1]
```diff
@@ -89,146 +96,219 @@ export class WalletProvider {
...
+     * @returns {Promise<number>} - Returns a promise that resolves to the current chainId.
+     */
     private async getWalletChain(): Promise<Chain> {
         if (!this.ethereum)
             throw new Error(Web3Errors.WALLET_NOT_CONNECTED)
 
-        return (await this.ethereum?.getNetwork())?.chainId as Chain
+        const chain = (await this.ethereum.getNetwork()).chainId
+
+        return parseChain(chain) as Chain
```
Comment (matheus-pissaia) [inline src/classes/WalletProvider/WalletProvider.ts:234]: `parseChain` should return a `Chain` type without the needing of enforcing with `as`

File: src/classes/WalletProvider/WalletProvider.ts [hunk 2]
```diff
@@ -64,22 +62,31 @@ export class WalletProvider {
...
         if (!provider)
             throw new Error(Web3Errors.PROVIDER_UNAVAILABLE)
 
-        this.ethereum = new ethers.providers.Web3Provider(provider, 'any') as ExtendedWeb3Provider
+        this.ethereum = new BrowserProvider(provider)
 
         if (!this.initializedEvents) {
             this._addEventListener('accountsChanged', this._updateAddress)
             this._addEventListener('chainChanged', this._updateChain)
+            // TODO handle disconnect event
```
Comment (matheus-pissaia) [inline src/classes/WalletProvider/WalletProvider.ts:84]: Are you going to resolve this TODO in the current PR?

File: src/chain/chains.ts [hunk 3]
```diff
@@ -17,9 +17,15 @@ export const allChains: Chain[] = Object.keys(Chain)
...
+        chain = chain.startsWith('0x')
+            ? parseInt(chain, 16) // If it's hex, parse it as base 16
+            : Number(chain) // If it's not hex, convert it to Number directly
+    }
+    else if (typeof chain === 'bigint') {
+        chain = Number(chain)
+    }
 
     if (Object.values(Chain).includes(chain))
         return chain as Chain
```
[the remark this answers is not in the document]
Comment (matheus-pissaia) [reply src/chain/chains.ts:31]: Also, you can reuse this method in other places and maybe add a second parameter like a custom chain array

File: src/classes/WalletProvider/WalletProvider.ts (hunk 2 above)
Comment (RauppRafael) [reply src/classes/WalletProvider/WalletProvider.ts:84]: no need for this event

**Review by iankressin — CHANGES_REQUESTED** [review]

**Review by iankressin — APPROVED** [review]

**Review by iankressin — APPROVED** [review]

**Review by matheus-pissaia — APPROVED** [review]

**Pull request merged**